packages/application-components/src/components/detail-pages/custom-form-detail-page/custom-form-detail-page.tsx

```diff
@@ -0,0 +1,145 @@
+import type { ReactElement, ReactNode, MouseEvent, KeyboardEvent } from 'react';
+import { sharedMessages } from '@commercetools-frontend/i18n';
+import { warning } from '@commercetools-uikit/utils';
+import styled from '@emotion/styled';
+import { customProperties } from '@commercetools-uikit/design-system';
+import {
+  FormPrimaryButton,
+  FormSecondaryButton,
+  FormDeleteButton,
+} from '../../internals/default-form-buttons';
+import Spacings from '@commercetools-uikit/spacings';
+import PageHeaderTitle from '../../internals/page-header-title';
+import PageTopBar from '../../internals/page-top-bar';
+import { ContentWrapper, PageWrapper } from '../../internals/page.styles';
+
+const noop = () => {};
+
+const DetailPageContainer = styled.div`
+  background-color: ${customProperties.colorNeutral95};
+  padding: ${customProperties.spacingM};
+  border-bottom: 1px ${customProperties.colorNeutral} solid;
+  display: flex;
+  align-items: flex-end;
+  justify-content: space-between;
+`;
+
+// NOTE: the `MessageDescriptor` type is exposed by `react-intl`.
+// However, we need to explicitly define this otherwise the prop-types babel plugin
+// does not recognize the object shape.
+type MessageDescriptor = {
+  id: string;
+  description?: string | object;
+  defaultMessage?: string;
+  values?: Record<string, ReactNode>;
+};
+
+type CustomFormDetailPageProps = {
+  /**
+   * The title of the page.
+   */
+  title?: string;
+  /**
+   * The subtitle of the page.
+   */
+  subtitle?: string | ReactElement;
+  /**
+   * Replaces the title/subtitle row with a custom one (for special use cases)
+   */
+  customTitleRow?: ReactNode;
+  /**
+   * Any React node displayed as the content within the page.
+   */
+  children: ReactNode;
+
+  // Controls Props
+  /**
+   * Any React node to be rendered as the form controls
+   */
+  formControls?: ReactNode;
+  /**
+   * Determines if the form controls should be rendered.
+   */
+  hideControls?: boolean;
+
+  // PageTopBar props:
+  /**
+   * Makes page top bar visible
+   */
+  showPageTopBar?: boolean;
```

Comment:
Is this a new prop? What's the purpose of it? 🤔

Reply:
Thanks. Looking at what is expected at `/customers/new` in the MC:
<img width="1392" alt="Screenshot 2022-05-09 at 17 29 05" src="https://user-images.githubusercontent.com/49066275/167595327-6461dbd8-e614-4111-961a-8e81e94040e3.png">
the top bar has to be optional.

The other though I had was to render it optionally based on `onPreviousPathClick`

Reply:
> The other though I had was to render it conditionally based on the `onPreviousPathClick`

Yeah maybe that's better so we don't need an extra prop for that.